File: test_db.py
import sqlite3

def dump_schema():
    conn = sqlite3.connect(r"d:\App\kaalakkani\assets\db\kaalakkani.db")
    cursor = conn.cursor()
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    for table_name in tables:
        name = table_name[0]
        print(f"--- Table: {name} ---")
        cursor.execute(f"PRAGMA table_info({name});")
        columns = cursor.fetchall()
        for col in columns:
            print(f"  Col: {col[1]} ({col[2]})")
        
        # sample a row
        cursor.execute(f"SELECT * FROM {name} LIMIT 1;")
        row = cursor.fetchone()
        print(f"  Sample row: {row}")
        print()
    conn.close()

if __name__ == "__main__":
    dump_schema()

File: test_db.js
const { DatabaseSync } = require('node:sqlite');
const path = require('node:path');

function dump() {
  const dbPath = path.resolve('d:/App/kaalakkani/assets/db/kaalakkani.db');
  console.log('Opening DB:', dbPath);
  const db = new DatabaseSync(dbPath);

  // Get all tables
  const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table';").all();
  for (const t of tables) {
    const name = t.name;
    console.log(`--- Table: ${name} ---`);
    try {
      const columns = db.prepare(`PRAGMA table_info(${name});`).all();
      for (const col of columns) {
        console.log(`  Col: ${col.name} (${col.type})`);
      }
      const row = db.prepare(`SELECT * FROM ${name} LIMIT 1;`).all();
      console.log('  Sample row:', JSON.stringify(row[0], null, 2));
    } catch (e) {
      console.error(e);
    }
    console.log();
  }
}

dump();
